Warn against enabling compile time filters in the readme

rukai opened the pull request:
closes https://github.com/tokio-rs/console/issues/315

rukai committed:
Warn against enabling compile time filters in the readme

hawkw commented on README.md:
```diff
@@ -116,14 +116,16 @@ notes:
   [`EnvFilter`] or [`Targets`] filters from [`tracing-subscriber`], add
   `"tokio=trace,runtime=trace"` to your filter configuration.
 
+  + also ensure you have not enabled any of the [compile time filter features](https://docs.rs/tracing/latest/tracing/level_filters/index.html#compile-time-filters) in your `Cargo.toml`.
```
```suggestion
  + also, ensure you have not enabled any of the [compile time filter features](https://docs.rs/tracing/latest/tracing/level_filters/index.html#compile-time-filters) in your `Cargo.toml`.
```

hawkw reviewed (changes requested):
Thanks, this looks good to me. I think we should also add a similar note to the `console-subscriber` crate's README, here: https://github.com/tokio-rs/console/blob/9178ecf02f094f8b23dc26f02faaba4ec09fd8f5/console-subscriber/README.md?plain=1#L55-L91 which has more detailed instructions on how to set up the console's instrumentation

rukai committed:
review feedback

rukai commented:
I copied over the whole section because that readme completely lacked that section, does this look good?

rukai commented:
I noticed and fixed some linter errors.
I also noticed the link to `build()` was dead so I replaced it with a link to Builder, hope that is still correct.

hawkw commented on README.md:
```diff
@@ -249,7 +251,8 @@ cargo run --example $name
 [`TRACE` level]: https://docs.rs/tracing/latest/tracing/struct.Level.html#associatedconstant.TRACE
-[build]: https://tokio-console.netlify.app/console_subscriber/fn.build.html
+[builder]: https://tokio-console.netlify.app/console_subscriber/struct.builder.html
 [init]: https://tokio-console.netlify.app/console_subscriber/fn.init.html
```
these linked to the Netlify documentation for the `main` branch because the `console-subscriber` crate had not been published to creates.io at the time the README was written. now, though, `console-subscriber` is published, so we should change these to link to docs.rs instead:
```suggestion
[builder]: https:/docs.rs/console-subscriber/latest/console_subscriber/struct.builder.html
[init]: https:/docs.rs/console-subscriber/latest/console_subscriber/fn.init.html
```

hawkw commented on console-subscriber/README.md:
```diff
@@ -123,6 +136,11 @@ Other instrumentation is added in later Tokio releases:
+[`EnvFilter`]: https://docs.rs/tracing-subscriber/latest/tracing_subscriber/filter/struct.EnvFilter.html
+[`Targets`]: https://docs.rs/tracing-subscriber/latest/tracing_subscriber/filter/targets/struct.Targets.html
+[builder]: https://tokio-console.netlify.app/console_subscriber/struct.builder.html
+[init]: https://tokio-console.netlify.app/console_subscriber/fn.init.html
```
similarly, let's have this link to docs.rs:
```suggestion
[builder]: https:/docs.rs/console-subscriber/latest/console_subscriber/struct.builder.html
[init]: https:/docs.rs/console-subscriber/latest/console_subscriber/fn.init.html
```
```

hawkw commented on README.md:
```diff
@@ -116,14 +116,16 @@ notes:
   [`EnvFilter`] or [`Targets`] filters from [`tracing-subscriber`], add
   `"tokio=trace,runtime=trace"` to your filter configuration.
 
+  + also, ensure you have not enabled any of the [compile time filter features][compile_time_filters] in your `Cargo.toml`.
```
mind wrapping this line at the same column as the rest of the bullet points?
```suggestion
  + also, ensure you have not enabled any of the [compile time filter 
    features][compile_time_filters] in your `Cargo.toml`.
```

hawkw commented on console-subscriber/README.md:
```diff
@@ -89,6 +89,19 @@ runtime][Tokio] is considered *experimental*. In order to use
+  [`EnvFilter`] or [`Targets`] filters from [`tracing-subscriber`], add
+  `"tokio=trace,runtime=trace"` to your filter configuration.
+
+  + also, ensure you have not enabled any of the [compile time filter features][compile_time_filters] in your `Cargo.toml`.
```
mind wrapping this line at the same column as the rest of the bullet points?
```suggestion
  + also, ensure you have not enabled any of the [compile time filter 
    features][compile_time_filters] in your `Cargo.toml`.
```

hawkw reviewed (commented):
i had some minor suggestions. besides that, this looks good to me, thank you!

hawkw committed:
Apply suggestions from code review

hawkw commented on README.md:
```diff
@@ -249,7 +252,8 @@ cargo run --example $name
 [`Layer`]: https://docs.rs/tracing-subscriber/latest/tracing_subscriber/layer/trait.Layer.html
 [`tracing` target]: https://docs.rs/tracing/latest/tracing/struct.Metadata.html
```
this link is broken:
```suggestion
[`tracing` targets]: https://docs.rs/tracing/latest/tracing/struct.Metadata.html
```

hawkw commented on console-subscriber/README.md:
```diff
@@ -123,6 +137,11 @@ Other instrumentation is added in later Tokio releases:
 [`tokio::time`]: https://docs.rs/tokio/1/tokio/time/index.html
 [Tokio v1.15.0]: https://github.com/tokio-rs/tokio/releases/tag/tokio-1.13.0
 [`tokio::sync`]: https://docs.rs/tokio/1/tokio/sync/index.html
+[`EnvFilter`]: https://docs.rs/tracing-subscriber/latest/tracing_subscriber/filter/struct.EnvFilter.html
```
noticed some broken links here:
```suggestion
[`tracing` targets]: https://docs.rs/tracing/latest/tracing/struct.Metadata.html
[`TRACE` level]: https://docs.rs/tracing/latest/tracing/struct.Level.html#associatedconstant.TRACE
[`EnvFilter`]: https://docs.rs/tracing-subscriber/latest/tracing_subscriber/filter/struct.EnvFilter.html
```

hawkw committed:
Fix broken links

hawkw commented on console-subscriber/README.md:
```diff
@@ -89,6 +89,20 @@ runtime][Tokio] is considered *experimental*. In order to use
+  `"tokio=trace,runtime=trace"` to your filter configuration.
+
+  + also, ensure you have not enabled any of the [compile time filter 
+    features][compile_time_filters] in your `Cargo.toml`.
```
The rest of the bullet points in this file are capitalized, so let's capitalize these for consistency:
```suggestion
* The `tokio` and `runtime` [`tracing` targets] must be enabled at the [`TRACE`
  level].

  + If you're using the [`console_subscriber::init()`][init] or
  [`console_subscriber::Builder`][builder] APIs, these targets are enabled
  automatically.

  + If you are manually configuring the `tracing` subscriber using the
  [`EnvFilter`] or [`Targets`] filters from [`tracing-subscriber`], add
  `"tokio=trace,runtime=trace"` to your filter configuration.

  + Also, ensure you have not enabled any of the [compile time filter 
    features][compile_time_filters] in your `Cargo.toml`.
```

hawkw committed:
Capitalize bullet points in console-subscriber/README.md

hawkw reviewed (approved):
this looks good to me, thanks for the PR!

hawkw commented:
CI failure is unrelated, looks like Clippy added new lints in Rust 1.60: https://github.com/tokio-rs/console/runs/5874325383?check_suite_focus=true

I'm going to fix this in a separate branch and then merge `main` into this PR.

hawkw committed:
Merge branch 'main' into improve_readme
